Document.pageContent should not allow undefined

### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain.js documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain.js rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

```typescript
import { describe, it, expect } from 'vitest'
import { Document } from '@langchain/core/documents'

describe('langchain Type Compatibility Test', () => {
  it('Document should handle empty pageContent', () => {
    const doc = new Document({ pageContent: '' })
    expect(doc.pageContent).toStrictEqual('')
  })
})
```

### Error Message and Stack Trace (if applicable)

AssertionError: expected undefined to strictly equal ''
Expected :
Actual   :undefined

### Description

pageContent should be set to '', as mandated by its type contract `string` (and not `string|undefined`)

The bug is visible in the code
 https://github.com/langchain-ai/langchainjs/blob/main/langchain-core/src/documents/document.ts
```typescript
export class Document<
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  Metadata extends Record<string, any> = Record<string, any>
> implements DocumentInput, DocumentInterface
{
  pageContent: string;

  metadata: Metadata;

  constructor(fields: DocumentInput<Metadata>) {
    this.pageContent = fields.pageContent
      ? fields.pageContent.toString()
      : this.pageContent;
    this.metadata = fields.metadata ?? ({} as Metadata);
  }
}
```

Line 32 should check field.pageContent is undefined, not just truthy, otherwise it uses this.pageContent which is undefined and breaks the contract that pageContent must be a string. Thus in any event, pageContent needs to be initialised to something to fulfill the type contract

### System Info

This will happen on any environment

--- a/langchain-core/src/documents/document.ts
+++ b/langchain-core/src/documents/document.ts
@@ -20,8 +20,8 @@ export interface DocumentInterface<
  * Interface for interacting with a document.
  */
 export class Document<
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  Metadata extends Record<string, any> = Record<string, any>
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    Metadata extends Record<string, any> = Record<string, any>
 > implements DocumentInput, DocumentInterface
 {
   pageContent: string;
@@ -29,9 +29,9 @@ export class Document<
   metadata: Metadata;
 
   constructor(fields: DocumentInput<Metadata>) {
-    this.pageContent = fields.pageContent
+    this.pageContent = fields.pageContent !== undefined
       ? fields.pageContent.toString()
-      : this.pageContent;
+      : "";
     this.metadata = fields.metadata ?? ({} as Metadata);
   }
 }

--- a/langchain-core/src/documents/document.ts
+++ b/langchain-core/src/documents/document.ts
@@ -20,8 +20,8 @@ export interface DocumentInterface<
  * Interface for interacting with a document.
  */
 export class Document<
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    Metadata extends Record<string, any> = Record<string, any>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  Metadata extends Record<string, any> = Record<string, any>
 > implements DocumentInput, DocumentInterface
 {
   pageContent: string;
@@ -29,9 +29,8 @@ export class Document<
   metadata: Metadata;
 
   constructor(fields: DocumentInput<Metadata>) {
-    this.pageContent = fields.pageContent !== undefined
-      ? fields.pageContent.toString()
-      : "";
+    this.pageContent =
+      fields.pageContent !== undefined ? fields.pageContent.toString() : "";
     this.metadata = fields.metadata ?? ({} as Metadata);
   }
 }

--- a/langchain-core/src/tests/document.test.ts
+++ b/langchain-core/src/tests/document.test.ts
@@ -1,7 +1,7 @@
-import {test, expect} from '@jest/globals'
-import {Document} from '../documents/document.js'
+import { test, expect } from "@jest/globals";
+import { Document } from "../documents/document.js";
 
-test('Document should handle empty pageContent', () => {
-  const doc = new Document({pageContent: ''})
-  expect(doc.pageContent).toEqual('')
-})
+test("Document should handle empty pageContent", () => {
+  const doc = new Document({ pageContent: "" });
+  expect(doc.pageContent).toEqual("");
+});

--- a/langchain-core/src/tests/document.test.ts
+++ b/langchain-core/src/tests/document.test.ts
@@ -1,0 +1,7 @@
+import {test, expect} from '@jest/globals'
+import {Document} from '../documents/document.js'
+
+test('Document should handle empty pageContent', () => {
+  const doc = new Document({pageContent: ''})
+  expect(doc.pageContent).toEqual('')
+})

